src/platform/packages/shared/kbn-scout/src/playwright/utils/a11y_utils.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { Page } from '@playwright/test';
+
+// selector should be on the page and match at least one element
+const verifyValidSelector = async (selector: string, page: Page): Promise<void> => {
+  const matchingElements = await page.locator(selector).count();
+  if (matchingElements === 0) throw new Error(`No elements found with selector: ${selector}`);
+};
+
+const keyToElement = async (
+  page: Page,
+  selector: string,
+  key: string,
+  visited: Set<string> = new Set()
+): Promise<void> => {
+  await page.keyboard.press(key);
+
+  const focusedOuterHTML = await page.evaluate(() => {
+    const focused = document.activeElement;
+    if (!focused) throw new Error('No focusable elements found on the page.');
+    return focused.outerHTML;
+  });
+  if (visited.has(focusedOuterHTML))
+    throw new Error(
```

Comment:
Could you check if it would work for virtually scrolled elements, which could be attached or detached from DOM depending on the scroll.

I suspect if reference changes after reattachment, this circular check may not work. If it's the case, see if it makes sense to add a `maxElementsToTraverse` guard.

Discover table is virtually scrolled in paged mode, so it can be tested for the purpose.

https://github.com/user-attachments/assets/5fb6fda6-625c-4f28-a33b-0c49e5ff9dc3

Reply:
`1000 ` looks like a fitting default for focus traversing. Great to know you've checked the failure scenario when the limit hits.